### ISSUE TITLE
Setting API timeout

### ISSUE THREAD

**Issue by blaind** [rackspace/php-opencloud#643]
I've been looking for a way to set API-call timeout (for cases when openstack goes down - default is curl default, over 30 secs). Is this the correct way to set the timeout?

    $os = new OpenStack(...);
    
    // set connection options (timeout)
    $config = $os->getConfig();
    $curl_config = $config->get('curl.options');
    $curl_config[CURLOPT_TIMEOUT] = X;
    $curl_config[CURLOPT_CONNECTTIMEOUT] = X;
    $config->set('curl.options', $curl_config);
    $os->setConfig($config);
    
    $service = $os->objectStoreService("swift", ...);

**Comment by jamiehannaford** [comment]
@blaind Yes, that looks correct. Does it not work for you?

**Comment by blaind** [comment]
Yes, it works. There was no mention about timeout in docs or past issues, so this works as a future reference for someone with the same problem. 

Maybe there could be an easier way to set curl options? E.g. by supplying an array of params in the OpenStack constructor?
